lib/nokogiri/html/document_fragment.rb

```diff
@@ -33,7 +33,7 @@ def initialize document, tags = nil, ctx = nil
           self.errors = document.errors - preexisting_errors
         else
           # This is a horrible hack, but I don't care
-          if tags.strip =~ /^<body/i
+          if /^\s*?<body/i.match?(tags)
```

Comment:
Can you tell me why you've inserted `\s*` at the front of this? That's a functional change that I wouldn't expect based on your description; and there aren't any tests provided demonstrating why this is introduced.

Reply:
The original code `tags.strip =~ /^<body/i` removes whitespace around string `tags` and tests if `tags` starts with `<body`.
So I thought instead of allocating a sanitized version of `tags` and testing, it'd be better if test was to check if `<body` substring exists preceded by an optional amount of whitespace from the start of string.

In other words, to do away with `tags.strip`...